strapi/strapi
# feat(cli): cloud cli commands (v5)

### What does it do?

This pull request introduces new commands to the Strapi CLI for interacting with Strapi Cloud:

- `strapi login`: This command allows users to log in to their Strapi Cloud application directly from the command line.
- `strapi logout`: This command enables users to log out of their currently authenticated Strapi Cloud session.
- `strapi deploy`: This command provides a streamlined way to deploy the local Strapi project directly to Strapi Cloud.
These functionalities aim to enhance the developer experience by simplifying authentication and deployment workflows for Strapi Cloud users and providing a seamless hosting solution for Strapi CMS users.

 ⚠️ As an MVP restriction, for now, only trial project can be created and managed from the deploy command ⚠️

### Why is it needed?

*Streamlining deployments*: The deploy command automates the deployment process, allowing developers to push their local project to Strapi Cloud directly from the CLI.

### How to test it?

1. Login: Use the `strapi login` command to authenticate with your Strapi Cloud account. You'll likely be prompted for your credentials during this process (Google / GitHub or GitLab login providers ATM).
2. Deploy: Ensure you have a local Strapi project ready, you must be able to deploy a trial project (never had any project on Strapi Cloud). Use the `strapi deploy` command to deploy your project to Strapi Cloud. Verify that the deployment process was completed successfully and your application is accessible on Strapi Cloud.
3. Logout: User the `strapi logout` command to disconnect the CLI from your Strapi Cloud account. The `strapi deploy` should not work after being logged out.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
